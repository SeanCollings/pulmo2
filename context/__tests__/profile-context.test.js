import React, { useContext } from 'react';
import { create } from 'react-test-renderer';
import { cleanup } from 'react-native-testing-library';

import ProfileContextProvider, { ProfileContext } from '../profile-context';

const MockComponent = () => {
  const context = useContext(ProfileContext);
  expect(context).toMatchSnapshot();
  return null;
};

describe('profile-context', () => {
  afterEach(cleanup);

  test('unit test', () => {
    let context = create(
      <ProfileContextProvider>
        <MockComponent />
      </ProfileContextProvider>
    );

    expect(context).toMatchSnapshot();
  });
});
