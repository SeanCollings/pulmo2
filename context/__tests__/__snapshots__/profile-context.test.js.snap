// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile-context unit test 1`] = `
Object {
  "profileContext": Object {
    "disclaimer": Object {
      "disclaimer_show": true,
      "dont_show_again": false,
    },
    "selected_level": 5,
    "streak": Object {
      "current": 0,
      "lastActivity": "",
      "streak_show": true,
    },
    "work_average_deviation": Object {
      "current": 0,
      "improvement": "same",
      "total_activities": 0,
    },
  },
  "updateCurrentStreak": [Function],
  "updateProfileContext": [Function],
  "updateWorkAverageDeviation": [Function],
}
`;

exports[`profile-context unit test 2`] = `null`;
